### PR TITLE
add e2e test for changing browser preference download directory

### PR DESCRIPTION
**Pull request by kuceb**
- add e2e test for changing default download directory in chrome/firefox

should be used in a docs example for mutating browser prefs

Docs PR: https://github.com/cypress-io/cypress-documentation/pull/2522

### CODE REVIEW

**Comment by jennifer-shehane** [comment]
Opened a PR to add example to docs: https://github.com/cypress-io/cypress-documentation/pull/2522

File: packages/server/test/e2e/4_before_browser_launch_spec.ts [hunk 1]
```diff
@@ -35,4 +35,10 @@ describe('e2e before:browser:launch', () => {
     sanitizeScreenshotDimensions: true,
     snapshot: true,
   })
+
+  e2e.it('can change browser prefs (download dir)', {
+    spec: '*',
```
Comment (flotwig) [inline packages/server/test/e2e/4_before_browser_launch_spec.ts]: there's no snapshot, and the spec is *, which means there's no assertion that the expected spec actually ran, just that there was some specfile, so i think you should add `snapshot: true` here

File: packages/server/test/support/fixtures/projects/browser-preferences/cypress/integration/spec.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+    link.setAttribute('href', encodedUri)
+    link.setAttribute('download', 'my_data.csv')
+    document.body.appendChild(link) // Required for FF
+
+    // This will download the data file named "my_data.csv".
+    link.click()
+
+    // ensure we downloaded to the correct directory.
+    cy.readFile('cypress/downloads/my_data.csv')
```
Comment (flotwig) [inline packages/server/test/support/fixtures/projects/browser-preferences/cypress/integration/spec.js]: think it's worth it to add an assertion that this contains `rows.map((e) => e.join(',')).join('\n')`?

**Review by flotwig — CHANGES_REQUESTED** [review]

File: packages/server/test/e2e/4_before_browser_launch_spec.ts (hunk 1 above)
Comment (Bkucera) [reply packages/server/test/e2e/4_before_browser_launch_spec.ts]: fair

File: packages/server/test/support/fixtures/projects/browser-preferences/cypress/integration/spec.js (hunk 2 above)
Comment (Bkucera) [reply packages/server/test/support/fixtures/projects/browser-preferences/cypress/integration/spec.js]: fair

**Comment by jennifer-shehane** [comment]
Closing this due to inactivity. Please reopen if there is work to be done addressing this again.

**Pull request closed without merging**